src/lexer/token.py

```diff
@@ -1,43 +1,130 @@
+from constants.constants import DELIMS
+from enum import Enum
+
+
+class TokenType(Enum):
+    def __init__(self, token: str, delim_id: str):
+        self._token = token
+        self._delim_id = delim_id
+        self._expected_delims = DELIMS[self.delim_id]
```

Comment:
Is this working fine?
```suggestion
        self._expected_delims = DELIMS[self._delim_id]
```